apps/andi/lib/andi_web/live/dataset_live_view.ex

```diff
@@ -25,27 +25,53 @@ defmodule AndiWeb.DatasetLiveView do
           </div>
         </form>
       </div>
-      <%= live_component(@socket, Table, datasets: @datasets) %>
+      <%= live_component(@socket, Table, id: :datasets_table, datasets: @datasets, order: @order) %>
     </div>
     """
   end
 
   def mount(_session, socket) do
-    {:ok, assign(socket, datasets: nil, search_text: "")}
+    {:ok, assign(socket, datasets: nil, search_text: nil, order: {"data-title", "asc"}, params: %{})}
+  end
+
+  def handle_info({:order, field}, socket) do
+    order_dir =
+      case socket.assigns.order do
+        %{^field => "asc"} -> "desc"
+        _ -> "asc"
+      end
+
+    params = Map.merge(socket.assigns.params, %{"order-by" => field, "order-dir" => order_dir})
+
+    {:noreply, live_redirect(socket, to: Routes.live_path(socket, __MODULE__, params))}
   end
 
   def handle_params(params, _uri, socket) do
+    order_by = Map.get(params, "order-by", "data-title")
+    order_dir = Map.get(params, "order-dir", "asc")
     search_text = Map.get(params, "search-value", "")
-    datasets = Andi.DatasetCache.get_datasets()
 
-    {:noreply, assign(socket, search_text: search_text, datasets: filter_datasets(datasets, search_text))}
+    datasets =
+      filter_on_search_change(search_text, socket)
+      |> sort_by_dir(order_by, order_dir)
+
+    {:noreply,
+     assign(socket, search_text: search_text, datasets: datasets, order: %{order_by => order_dir}, params: params)}
```

Comment:
Did we confirm that the order part is not additive? That is, if we change the column that the other one doesn't hang around in the map? I'm 90% sure it kills other entries (like it should), but just double checking

Reply:
Yep, otherwise the css indicator for sorting would show on both columns when you switch between them